File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/producer/fallback/MessageClient.java
package io.micronaut.configuration.kafka.docs.producer.fallback;

import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.configuration.kafka.annotation.Topic;

@KafkaClient
public interface MessageClient {

    @Topic("messages")
    void send(String message);
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/fallback/MessageClient.py
from abc import ABC, abstractmethod

from micronaut.configuration.kafka.annotation import KafkaClient, Topic
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="MessageClientFallbackSpec")
@KafkaClient
class MessageClient(ABC):

    @Topic("messages")
    @abstractmethod
    def send(self, message: str) -> None:
        ...
